Develop1

smaikles opened the pull request:
Задание 1: юнит-тесты

Commit:
- создал ветку develop1
- добавил зависимости в pom файл

Commit:
- создал тестовые классы по заданию
- написал тест для класса Bug
- добавил readme.md

Commit:
- написал тесты для остальных классов
- добавил отчет Jacoco

pomegranate282922 commented on pom.xml:
```diff
@@ -5,12 +5,55 @@
 
+    <dependencies>
+        <dependency>
+            <groupId>junit</groupId>
```
Отлично! Ничего лишнего!

pomegranate282922 commented on src/test/java/BunTest.java:
```diff
@@ -0,0 +1,52 @@
+        this.price = price;
+    }
+
+    @Parameterized.Parameters
```
Отлично! Протестированы разные значения

pomegranate282922 commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,119 @@
+    Burger burger;
+
+    @Mock
+    Bun mockBun;
```
Отлично! Моки применены уместно.

pomegranate282922 commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,119 @@
+        burger.setBuns(mockBun);
+        Float actual = burger.getPrice();
+        Float expected = 246.8f;
+        assertEquals("Цена должна быть",expected,actual,0.01f);
```
Лучше воспользоваться форматированием в IDEA - не хватает пробелов.

pomegranate282922 commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,119 @@
+        burger.addIngredient(ingredient);
+        int ingredientSize = burger.ingredients.size();
+        boolean actual = ingredientSize > 0;
+        assertTrue("Кол-во ингридиентов должно быть > 0", actual);
```
Нужно более четкое требование сформулировать. Зачем такое неопределённое?

pomegranate282922 commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,119 @@
+        Mockito.when(mockBun.getPrice()).thenReturn(123.4f);
+        burger.setBuns(mockBun);
+        Float actual = burger.getPrice();
+        Float expected = 246.8f;
```
Не полностью протестирован метод цены. Нужно дополнить.

pomegranate282922 commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,119 @@
+        burger.setBuns(bun);
+        buns.add(bun);
+        int bunsSize = buns.size();
+        boolean actual = bunsSize > 0;
```
Здесь тоже нужно определеннее условие сделать

pomegranate282922 commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,119 @@
+    public void checkBunsSet() {
+        List<Bun> buns = new ArrayList<>();
+        Bun bun = Mockito.mock(Bun.class);
+        Burger burger = new Burger();
```
Не нужно новый объект и мок создавать - нужно использовать поля класса, раз они уже есть

pomegranate282922 commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,119 @@
+
+    @Test
+    public void checkDeleteIngredients() {
+        Burger burger = new Burger();
```
Аналогично - не нужен новый объект

pomegranate282922 commented on src/test/java/Ingredientest.java:
```diff
@@ -0,0 +1,2 @@
+public class Ingredientest {
```
Пустые классы нужно удалить

pomegranate282922 reviewed (commented):
Работа хорошая - тесты все на месте, параметризация и моки применены уместно.
Нужно внести изменения согласно замечаниям.

Commit:
Выполнил работу над замечаниями:
- сделал выравнивание, применив форматирование IDEA
- обновил метод проверки цены
- дал более понятое определение ожидаемому результату
- поправил тест, убрав лишние обекты

Commit:
Дополнительно:
- перегенерировал отчет jacoco
- добавил в pom игнорирование не тестируемых классов
- убрал пустой класс

Commit:
Дополнительно:
- поправил, убрал пробелы и лишние новые обьекты, так как он уже обьявлен

pomegranate282922 commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,119 @@
+        int ingredientSize = burger.ingredients.size();
+        boolean actual = ingredientSize > 0;
+
+        assertTrue("В собранном бургере отсутсвуют ингридиенты, добавь хотябы 1", actual);
```
Немного не то имела в виду. 
по логике теста ты знаешь, что размер равен 1. Зачем проверять, что размер больше 0? Ведь если размер будет 10,  тест пройдёт, а баг будет пропущен.

smaikles replied:
Ох, да точно!!! я делал исправление поздно ночью и уже голова не варила :) Спасибо!

pomegranate282922 reviewed (approved):
Оставила комментарий, но в целом отлично - зачёт!
